app/tests/api/api_v1/test_locations.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+from httpx import AsyncClient
+
+from app.tests.client.db_client import MSSQLClient
+from app.tests.utils.locations_utils import LocationsDBHelper
+
+
+@pytest.mark.asyncio
+async def test_GivenNoLocations_WhenGettingLocations_ThenNothingReceived(
+    db_client: MSSQLClient, app_client: AsyncClient
+) -> None:
+    locations_helper = LocationsDBHelper(db_client)
+
+    response = await app_client.get("/api/v1/locations")
+
+    assert response.status_code == 200
+
+    content = response.json()
+    assert content == []
+
+
+@pytest.mark.asyncio
+async def test_GivenOneLocation_WhenGettingLocations_ThenLocationReceived(
+    db_client: MSSQLClient, app_client: AsyncClient
+) -> None:
+    locations_helper = LocationsDBHelper(db_client)
+
+    await locations_helper.create_location("Location1", 1)
+
+    response = await app_client.get("/api/v1/locations")
+
+    assert response.status_code == 200
+
+    content = response.json()
+    assert not content == []
```

Comment:
I'm gonna add more to this so it checks more stuff, this is not sufficient. Just lazy atm

Reply:
I've explained this in the intro for the PR, but in case you missed it, basically you can't have `async` tests if you use the regular `TestClient`. FastAPI explicitly suggests `AsyncClient` from `httpx` instead in their [docs](https://fastapi.tiangolo.com/advanced/async-tests/) if you want to have `async`able test cases.

I understand that `httpx` is technically in beta, but they do also mention that as of `0.18.*` things should be stable for the public, and a more general release is coming this year. And for the purposes of test code and to be able to use `async` freely, I wouldn't say it's a big issue. If it were part of the app, I might have just swallowed the pill and write stuff without `async` even if the code itself turned out more asinine.

Let me know if you still think we should `TestClient` instead, so then I can go and change everything to not be async, or if you have alternative suggestions.

Reply:
It's fine, I trust your judgement.